src/ui/components/filters/geolocationcomponent.js

```diff
@@ -119,10 +119,14 @@ export default class GeoLocationComponent extends Component {
      * The query string to use for the input box, provided to template for rendering.
      * @type {string}
      */
-    this.query = this.core.globalStorage.getState(`${StorageKeys.QUERY}.${this.name}`) || '';
-    this.core.globalStorage.on('update', `${StorageKeys.QUERY}.${this.name}`, q => {
-      this.query = q;
-      this.setState();
+    this.query = this.core.storage.get(`${StorageKeys.QUERY}.${this.name}`) || '';
```

Comment:
This might be an existing SDK idiosyncrasy, but is the `.${this.name} ` suffix here ever actually present? A cursory glance through the rest of this PR makes it seem like we're never setting `${StorageKeys.QUERY}.${this.name}`, only `${StorageKeys.QUERY}`.

Reply:
I think these come from persistentStorage, from when we seed globalStorage with persistentStorage on page load

Reply:
the new storage also has this seeding behavior it its init()!

Reply:
Ah, okay!